app/models/submission_information_package_zipper.rb

```diff
@@ -19,7 +19,7 @@ def initialize(sip)
   # This key needs to be unique. By default, ActiveStorage generates a UUID, but since we want a file path for our
   # Archivematica needs, we are generating a key. We handle uniqueness on the `bag_name` side.
   def keygen(sip)
-    "etdsip/#{sip.thesis.graduation_year}/#{sip.thesis.graduation_month}/#{sip.bag_name}.zip"
+    "etdsip/#{sip.thesis.accession_number}/#{sip.bag_name}.zip"
```

Comment:
The way `thesis.accession_number` works, it is possible for it to return `nil` in situations in which the an accession number was not created or the degree period somehow doesn't exist yet in the application.

I think our expectation is we won't allow publishing to happen and thus sip creation is impossible without valid accession numbers being available. Is that correct and do you think we should do something else here to protect against nil accession numbers?